packages/rs-platform-value/src/lib.rs

```diff
@@ -0,0 +1,963 @@
+//! Platform value
+//! A dynamic Platform value
+//!
+//! A module that is used to represent values in Platform components
+//! Forked from ciborium value
+//!
+//!
+pub mod btreemap_extensions;
+pub mod converter;
+pub mod display;
+mod error;
+pub mod inner_value;
+mod integer;
+pub mod system_bytes;
+pub mod value_map;
+
+pub use error::Error;
+pub use integer::Integer;
+
+pub type ValueMap = Vec<(Value, Value)>;
+pub type Hash256 = [u8; 32];
+
+/// A representation of a dynamic value that can handled dynamically
+#[non_exhaustive]
+#[derive(Clone, Debug, PartialEq, PartialOrd)]
+pub enum Value {
+    /// A u128 integer
+    U128(u128),
+
+    /// A i128 integer
+    I128(i128),
+
+    /// A u64 integer
+    U64(u64),
+
+    /// A i64 integer
+    I64(i64),
+
+    /// A u32 integer
+    U32(u32),
+
+    /// A i32 integer
+    I32(i32),
+
+    /// A u16 integer
+    U16(u16),
+
+    /// A i16 integer
+    I16(i16),
+
+    /// A u8 integer
+    U8(u8),
+
+    /// A i8 integer
+    I8(i8),
+
+    // Todo: add this in
+    // /// A 256 bit hash
+    // Hash256(Hash256),
+    /// Bytes
+    Bytes(Vec<u8>),
+
+    /// A float
+    Float(f64),
+
+    /// A string
+    Text(String),
+
+    /// A boolean
+    Bool(bool),
+
+    /// Null
+    Null,
+
+    /// Tag
+    Tag(u64, Box<Value>),
+
+    /// An array
+    Array(Vec<Value>),
+
+    /// A map
+    Map(ValueMap),
+}
+
+impl Value {
+    /// Returns true if the `Value` is an `Integer`. Returns false otherwise.
+    ///
+    /// ```
+    /// # use platform_value::Value;
+    /// #
+    /// let value = Value::U64(17);
+    ///
+    /// assert!(value.is_integer());
+    /// ```
+    pub fn is_integer(&self) -> bool {
+        match self {
+            Value::U128(_)
+            | Value::I128(_)
+            | Value::U64(_)
+            | Value::I64(_)
+            | Value::U32(_)
+            | Value::I32(_)
+            | Value::U16(_)
+            | Value::I16(_)
+            | Value::U8(_)
+            | Value::I8(_) => true,
+            _ => false,
+        }
+    }
+
+    /// If the `Value` is a `Integer`, returns a reference to the associated `Integer` data.
+    /// Returns None otherwise.
+    ///
+    /// ```
+    /// # use platform_value::Value;
+    /// #
+    /// let value = Value::U64(17);
+    ///
+    /// // We can read the number
+    /// let r_value : u64 = value.as_integer().unwrap();
+    /// assert_eq!(17, r_value);
+    /// ```
+    pub fn as_integer<T>(&self) -> Option<T>
```

Comment:
I see many conversions baked into `Value` type, but how about reducing methods' number and use a trait for it called `TryFrom`?

For instance, `as_` family could be done as `TryFrom<&'a Value> for sometype` and then `sometype::try_from(&value)` is available, for owning conversions `TryFrom<Value>` is also available.

Reply:
I thought about that, but I personally felt this was better, because otherwise you would need to type your variables a lot.

Like `let a : Option<String> = value.try_from()?`
compared to `let a = value.get_optional_string()?`

I personally prefer the second one, because you have the value, and then you express what you want from it.

I do think though it's probable a good idea to implement try_from's as well.